jakarta-eclipse/org.eclipse.lsp4jakarta.core/src/org/jakarta/jdt/jax_rs/RootResourceClassDiagnosticsCollector.java

```diff
@@ -0,0 +1,120 @@
+                        String className = type.getElementName();
+                        Map<IMethod, Integer> constructorParamsMap = new HashMap<IMethod, Integer>();
+                        
+                        for (IMethod method : type.getMethods()) {
+                            // if a method of a class has the same name as the class, it is a constructor
+                            if ((method.getElementName().equals(className)) && (Flags.isPublic(method.getFlags()))) {
+                                int numParams = method.getNumberOfParameters();
+                                
+                                if (numParams > maxParams) {
+                                    maxParams = numParams;
+                                }
+                                constructorParamsMap.put(method, numParams);
+                            }
+                        }
+                        
+                        ArrayList<IMethod> equalMaxParamMethods = new ArrayList<IMethod>();
+                        
+                        for (Map.Entry<IMethod, Integer> entry : constructorParamsMap.entrySet()) {
+                            if (entry.getValue() == maxParams) {
+                                equalMaxParamMethods.add(entry.getKey());
+                            }
+                            else if (entry.getValue() < maxParams) {
+                                IMethod method = entry.getKey();
+                                ISourceRange methodNameRange = JDTUtils.getNameRange(method); // TODO: maybe change the area this diagnostic underlines
+                                Range methodRange = JDTUtils.toRange(unit, methodNameRange.getOffset(), methodNameRange.getLength());
+                                
+                                diagnostic = new Diagnostic(methodRange, "This constructor is unused, as root resource classes will only use the constructor with the most parameters.");
+                                diagnostic.setSeverity(Jax_RSConstants.SEVERITY_WARNING);
+                                completeDiagnostic(diagnostic);
+                                diagnostics.add(diagnostic);
+                            }
+                        }
+                        if (equalMaxParamMethods.size() > 1) {
+                            for (IMethod method : equalMaxParamMethods) {
+                                ISourceRange methodNameRange = JDTUtils.getNameRange(method); // TODO: maybe change the area this diagnostic underlines
+                                Range methodRange = JDTUtils.toRange(unit, methodNameRange.getOffset(), methodNameRange.getLength());
+                                
+                                diagnostic = new Diagnostic(methodRange, "Multiple constructors have the same number of parameters, it may be ambiguous which constructor is used.");
+                                diagnostic.setSeverity(Jax_RSConstants.SEVERITY_ERROR);
```

Comment:
I think this should be a warning as well, to stay consistent and since it's not a breaking error. Let me know what you think @MShocrylas and @berasogut

Reply:
If we change this to a warning we can also remove explicitly setting the severities here and instead just call `completeDiagnostic` since you set the severity there as well.